Port out-of-line display list items to the new serialization format

gavin-apple opened the pull request:
#### 9dbbb78ff5f144ced7c9091bc09262d84df511d3
<pre>
Port out-of-line display list items to the new serialization format
<a href="https://bugs.webkit.org/show_bug.cgi?id=252895">https://bugs.webkit.org/show_bug.cgi?id=252895</a>
rdar://105881207

Reviewed by Alex Christensen.

Ports the out-of-line display list items to the new serialization format.
This includes:
    - SetState
    - SetLineDash
    - ClipOutToPath
    - ClipPath
    - ClipPath
    - DrawGlyphs
    - DrawSystemImage
    - DrawLinesForText
    - DrawPath
    - DrawFocusRingPath
    - DrawFocusRingRects
    - FillRectWithColor
    - FillRectWithGradient
    - FillCompositedRect
    - FillRoundedRect
    - FillRectWithRoundedHole
    - FillPath
    - StrokePath

* Source/WebCore/platform/graphics/displaylists/DisplayListItems.cpp:
(WebCore::DisplayList::DrawLinesForText::DrawLinesForText):
(WebCore::DisplayList::FillRectWithGradient::FillRectWithGradient):
* Source/WebCore/platform/graphics/displaylists/DisplayListItems.h:
(WebCore::DisplayList::DrawGlyphs::fontIdentifier const):
(WebCore::DisplayList::DrawGlyphs::positionedGlyphs const):
(WebCore::DisplayList::FillRectWithGradient::gradient const):
(WebCore::DisplayList::SetState::encode const): Deleted.
(WebCore::DisplayList::SetState::decode): Deleted.
(WebCore::DisplayList::SetLineDash::encode const): Deleted.
(WebCore::DisplayList::SetLineDash::decode): Deleted.
(WebCore::DisplayList::ClipOutToPath::encode const): Deleted.
(WebCore::DisplayList::ClipOutToPath::decode): Deleted.
(WebCore::DisplayList::ClipPath::encode const): Deleted.
(WebCore::DisplayList::ClipPath::decode): Deleted.
(WebCore::DisplayList::DrawGlyphs::fontIdentifier): Deleted.
(WebCore::DisplayList::DrawGlyphs::encode const): Deleted.
(WebCore::DisplayList::DrawGlyphs::decode): Deleted.
(WebCore::DisplayList::DrawSystemImage::encode const): Deleted.
(WebCore::DisplayList::DrawSystemImage::decode): Deleted.
(WebCore::DisplayList::DrawLinesForText::encode const): Deleted.
(WebCore::DisplayList::DrawLinesForText::decode): Deleted.
(WebCore::DisplayList::DrawPath::encode const): Deleted.
(WebCore::DisplayList::DrawPath::decode): Deleted.
(WebCore::DisplayList::DrawFocusRingPath::encode const): Deleted.
(WebCore::DisplayList::DrawFocusRingPath::decode): Deleted.
(WebCore::DisplayList::DrawFocusRingRects::encode const): Deleted.
(WebCore::DisplayList::DrawFocusRingRects::decode): Deleted.
(WebCore::DisplayList::FillRectWithColor::encode const): Deleted.
(WebCore::DisplayList::FillRectWithColor::decode): Deleted.
(WebCore::DisplayList::FillRectWithGradient::encode const): Deleted.
(WebCore::DisplayList::FillRectWithGradient::decode): Deleted.
(WebCore::DisplayList::FillCompositedRect::encode const): Deleted.
(WebCore::DisplayList::FillCompositedRect::decode): Deleted.
(WebCore::DisplayList::FillRoundedRect::encode const): Deleted.
(WebCore::DisplayList::FillRoundedRect::decode): Deleted.
(WebCore::DisplayList::FillRectWithRoundedHole::encode const): Deleted.
(WebCore::DisplayList::FillRectWithRoundedHole::decode): Deleted.
(WebCore::DisplayList::FillPath::encode const): Deleted.
(WebCore::DisplayList::FillPath::decode): Deleted.
(WebCore::DisplayList::StrokePath::encode const): Deleted.
(WebCore::DisplayList::StrokePath::decode): Deleted.
* Source/WebCore/platform/graphics/displaylists/DisplayListRecorderImpl.cpp:
(WebCore::DisplayList::RecorderImpl::recordDrawLinesForText):
* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/GPU/graphics/RemoteDisplayListRecorderProxy.cpp:
(WebKit::RemoteDisplayListRecorderProxy::recordDrawLinesForText):
* Tools/TestWebKitAPI/Tests/WebCore/DisplayListTests.cpp:
(TestWebKitAPI::TEST):

Canonical link: <a href="https://commits.webkit.org/260879@main">https://commits.webkit.org/260879@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5be11d25514cb0de9b0abd1332d0ce382f8796be

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/109469 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/18593 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/42231 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/967 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/118631 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/113354 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/20059 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/9790 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/101751 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/115224 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/14947 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/98170 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/43152 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/96919 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/29823 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/84914 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/11335 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/31166 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/11998 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/8097 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/17364 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/50770 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7546 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/13730 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebCore/platform/graphics/displaylists/DisplayListItems.h:
```diff
@@ -1207,81 +914,30 @@ class FillRectWithColor {
     static constexpr bool isInlineItem = false;
     static constexpr bool isDrawingItem = true;
 
     WEBCORE_EXPORT FillRectWithGradient(const FloatRect&, Gradient&);
```
We might be able to remove this constructor.

gavin-apple replied:
It's currently in use by our DisplayListRecorder

gavin-apple committed:
Port out-of-line display list items to the new serialization format
https://bugs.webkit.org/show_bug.cgi?id=252895
rdar://105881207

Reviewed by Alex Christensen.

Ports the out-of-line display list items to the new serialization format.
This includes:
    - SetState
    - SetLineDash
    - ClipOutToPath
    - ClipPath
    - ClipPath
    - DrawGlyphs
    - DrawSystemImage
    - DrawLinesForText
    - DrawPath
    - DrawFocusRingPath
    - DrawFocusRingRects
    - FillRectWithColor
    - FillRectWithGradient
    - FillCompositedRect
    - FillRoundedRect
    - FillRectWithRoundedHole
    - FillPath
    - StrokePath

* Source/WebCore/platform/graphics/displaylists/DisplayListItems.cpp:
(WebCore::DisplayList::DrawLinesForText::DrawLinesForText):
(WebCore::DisplayList::FillRectWithGradient::FillRectWithGradient):
* Source/WebCore/platform/graphics/displaylists/DisplayListItems.h:
(WebCore::DisplayList::DrawGlyphs::fontIdentifier const):
(WebCore::DisplayList::DrawGlyphs::positionedGlyphs const):
(WebCore::DisplayList::FillRectWithGradient::gradient const):
(WebCore::DisplayList::SetState::encode const): Deleted.
(WebCore::DisplayList::SetState::decode): Deleted.
(WebCore::DisplayList::SetLineDash::encode const): Deleted.
(WebCore::DisplayList::SetLineDash::decode): Deleted.
(WebCore::DisplayList::ClipOutToPath::encode const): Deleted.
(WebCore::DisplayList::ClipOutToPath::decode): Deleted.
(WebCore::DisplayList::ClipPath::encode const): Deleted.
(WebCore::DisplayList::ClipPath::decode): Deleted.
(WebCore::DisplayList::DrawGlyphs::fontIdentifier): Deleted.
(WebCore::DisplayList::DrawGlyphs::encode const): Deleted.
(WebCore::DisplayList::DrawGlyphs::decode): Deleted.
(WebCore::DisplayList::DrawSystemImage::encode const): Deleted.
(WebCore::DisplayList::DrawSystemImage::decode): Deleted.
(WebCore::DisplayList::DrawLinesForText::encode const): Deleted.
(WebCore::DisplayList::DrawLinesForText::decode): Deleted.
(WebCore::DisplayList::DrawPath::encode const): Deleted.
(WebCore::DisplayList::DrawPath::decode): Deleted.
(WebCore::DisplayList::DrawFocusRingPath::encode const): Deleted.
(WebCore::DisplayList::DrawFocusRingPath::decode): Deleted.
(WebCore::DisplayList::DrawFocusRingRects::encode const): Deleted.
(WebCore::DisplayList::DrawFocusRingRects::decode): Deleted.
(WebCore::DisplayList::FillRectWithColor::encode const): Deleted.
(WebCore::DisplayList::FillRectWithColor::decode): Deleted.
(WebCore::DisplayList::FillRectWithGradient::encode const): Deleted.
(WebCore::DisplayList::FillRectWithGradient::decode): Deleted.
(WebCore::DisplayList::FillCompositedRect::encode const): Deleted.
(WebCore::DisplayList::FillCompositedRect::decode): Deleted.
(WebCore::DisplayList::FillRoundedRect::encode const): Deleted.
(WebCore::DisplayList::FillRoundedRect::decode): Deleted.
(WebCore::DisplayList::FillRectWithRoundedHole::encode const): Deleted.
(WebCore::DisplayList::FillRectWithRoundedHole::decode): Deleted.
(WebCore::DisplayList::FillPath::encode const): Deleted.
(WebCore::DisplayList::FillPath::decode): Deleted.
(WebCore::DisplayList::StrokePath::encode const): Deleted.
(WebCore::DisplayList::StrokePath::decode): Deleted.
* Source/WebCore/platform/graphics/displaylists/DisplayListRecorderImpl.cpp:
(WebCore::DisplayList::RecorderImpl::recordDrawLinesForText):
* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/GPU/graphics/RemoteDisplayListRecorderProxy.cpp:
(WebKit::RemoteDisplayListRecorderProxy::recordDrawLinesForText):
* Tools/TestWebKitAPI/Tests/WebCore/DisplayListTests.cpp:
(TestWebKitAPI::TEST):

Canonical link: https://commits.webkit.org/260879@main
